plugins/link/dialogs/anchor.js

```diff
@@ -30,12 +30,38 @@ CKEDITOR.dialog.add( 'anchor', function( editor ) {
 			element = element.getParent();
 		}
 
+		// If anchor is exist and has some styles go to the closest parent <a> tag. (#3863)
+		if ( element && !element.is( 'a' ) ) {
+			element = element.getAscendant( 'a' ) || element;
+		}
+
 		if ( element && element.type === CKEDITOR.NODE_ELEMENT &&
 			( element.data( 'cke-real-element-type' ) === 'anchor' || element.is( 'a' ) ) ) {
 			return element;
 		}
 	}
 
+	// Checks if range contains some nested anchors and remove it.
+	function checkNestedAnchors( range ) {
```

Comment:
This functions removes all the anchors within the current range so the name is a little misleading. Maybe something like `removeAnchorsWithinRange()`?

Reply:
And with a descriptive name the comment above will not be needed.

Reply:
And with a descriptive name the comment above will not be needed.